testRearrangeCarGroup8.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import unittest
+import numpy as np
+from RearrangeCarGroup8Movements import RearrangeCars
+
+class TestRearrangeCars(unittest.TestCase):
+
+    '''Unittests for RearrangeCars class and promised fuctionality
+    '''
+
+    def setUp(self):
+
+        '''Basic setting that will be used in several tests
+        '''
+        self.init_order = [1, 2, 0, 3]
+        self.final_order = [3, 1, 2, 0]
+        self.rc = RearrangeCars(self.init_order, self.final_order)
+
+    def test_rearrange_all_cars(self):
+
+        '''Tests on the toy example without special difficulties
+        '''
+
+        self.rc.rearrange_all_cars()
+        self.assertEqual(self.rc.current_car_order, self.final_order)
+
+    def test_rearrange_all_cars_twice(self):
+
+        '''Test whether the class instance can properly handle
+        running the same function twice
+        '''
+
+        self.rc.rearrange_all_cars()
+        self.assertEqual(self.rc.current_car_order, self.final_order)
+        self.rc.rearrange_all_cars()
+        self.assertEqual(self.rc.current_car_order, self.final_order)
+
+    def test_already_proper_order(self):
+
+        '''Tests the case when initial and final orders are the same
+        '''
+
+        rc_same = RearrangeCars(self.final_order, self.final_order)
+        self.assertRaises(rc_same.rearrange_all_cars())

```

Comment:
It's nice to test against already ordered input! But since we are planning to remove the assert from the `rearrange_all_cars()`, we can probably change line 44 to `self.assertEqual(self.rc.current_car_order, self.final_order)` ^^